internal/events/api/event.go

```diff
@@ -37,7 +37,7 @@ type Sender interface {
 //go:generate mockgen -source=event.go -package=api -destination=mock_event.go
 type Handler interface {
 	Sender
-	V2GetEvents(ctx context.Context, clusterID *strfmt.UUID, hostID *strfmt.UUID, infraEnvID *strfmt.UUID, categories ...string) ([]*common.Event, error)
+	V2GetEvents(ctx context.Context, clusterID *strfmt.UUID, hostsID []strfmt.UUID, infraEnvID *strfmt.UUID, limit, offset *int64, severity []string, message *string, deletedHosts, clusterLevel *bool, categories ...string) ([]*common.Event, *common.EventSeverityCount, error)
```

Comment:
I would recommend trying to group arguments to have less in the signature.
This would allow for easier code reading and easier testing, making it less brittle: by not having to change the signature adding/removing/changing a filter would break hundreds of tests that do use the method but they don't rely on the filters.

Also in the response we can group them and allow for extensibility, for example returning a response where we could get events from:
```
response, err := events.V2GetEvents(...)
events := response.GetEvents()
severityCount := response.GetStats().GetSeverityCount()
```

with the above change, a call like:
```
evs, _, err := theEvents.V2GetEvents(context.TODO(), clusterID, hostsID, infraEnvID, common.UnlimitedEvents, common.NoOffsetEvents, nil, nil, nil, nil)
```
could be turned into something like:
```
eventFilters := events.DefaultEventFiltersWithIDs(clusterID, hostID, infraEnvID)
response, err := theEvents.V2GetEvents(context.TODO(), eventFilters)
evs := response.GetEvents()
```
and even when adding some other filters would still be very readable (and we could also have utility methods to return some pre-initialized common filters)
```
eventFilters := events.DefaultEventFiltersWithIDs(clusterID, hostID, infraEnvID)
eventFilters.Limit = 100
eventFilters.Offset = 20
response, err := theEvents.V2GetEvents(context.TODO(), eventFilters)
evs := response.GetEvents()
```